src/main/java/school/hei/haapi/endpoint/rest/mapper/GroupFlowMapper.java

```diff
@@ -0,0 +1,20 @@
+package school.hei.haapi.endpoint.rest.mapper;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import school.hei.haapi.endpoint.rest.model.GroupFlow;
+
+@Component
+@AllArgsConstructor
+public class GroupFlowMapper {
+
+  public GroupFlow toRest(school.hei.haapi.model.GroupFlow domain) {
+    return new GroupFlow()
+            .id(domain.getId())
+            .flowdDateTime(domain.getFlowDatetime())
+            .groupId(domain.getGroup().getId())
+            .studentId(domain.getStudent().getId())
+            .moveType(GroupFlow.MoveTypeEnum.valueOf(domain.getGroupFlowType().toString()));
```

Comment:
GroupFlow.MoveTypeEnum.valueOf(domain.getGroupFlowType().toString()) is deprecated.

Because you don't know what kind of value the consumer can provide. So, instead, compare each possible values of MoveTypeEnum and the  domain.getGroupFlowType() using switch - case. Default value throws unknown MoveType. Ask Vagno for more explanation.